akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGraphCompileSpec.scala

```diff
@@ -440,6 +440,32 @@ class FlowGraphCompileSpec extends AkkaSpec {
         }
       }
 
+      "Junction is connected through GraphFlow" in {
+        val gflow = Flow[Int, String]() { implicit builder ⇒
+          import FlowGraphImplicits._
+
+          val in = UndefinedSource[Int]
+          val out = UndefinedSink[String]
+
+          in ~> Flow[Int].map(_.toString) ~> out
+
+          (in, out)
```

Comment:
um, it would also avoid `in -> out`, which almost looks like `in ~> out`

first we must come up with a name for the case class, then we can consider the `Flow.apply` for creating it (I'm not convinced about the latter)